mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -344,6 +345,11 @@ def _transform_request_json_for_chat_if_necessary(request_json, lc_model):
                 2. A boolean indicating whether or not the request was transformed from the OpenAI
                 chat format.
         """
+        # Avoid converting the request to LangChain's Message format if the chain
```

Comment:
Can we add docstring about why we do this transformation and when it is `necessary`?  It's super hard to see what kind of input cases are expected and when the exception (like agent) applies....

Reply:
I think it's necessary for `ChatModel` to convert the input from dictionary format to ChatMessages, but for other model who originally accepts "messages" dict input it's not necessary.